src/components/ListItem.jsx

```diff
@@ -1,5 +1,59 @@
 import './ListItem.css';
+import { useState, useEffect } from 'react';
+import { updateItem } from '../api';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({ item, listToken }) {
+	const one_day_in_ms = 24 * 60 * 60 * 1000;
+	// const one_day_in_ms = 60 * 2 * 1000; // 120 seconds for testing the reset timeframe
+	const currentDate = new Date();
+	const currentTimeInMilliseconds = Math.floor(currentDate.getTime());
+	const dateLastPurchasedInMilliseconds =
+		item && item.dateLastPurchased
+			? item.dateLastPurchased.seconds * 1000
+			: null;
+	const timeElapsed =
+		currentTimeInMilliseconds - dateLastPurchasedInMilliseconds;
+
+	const [isPurchased, setIsPurchased] = useState(item.isChecked);
+
+	const handlePurchaseItem = async () => {
+		try {
+			if (item.isChecked === false) {
+				item.totalPurchases++;
+				item.isChecked = true;
+				await updateItem(listToken, item);
+			} else {
+				item.totalPurchases--;
+				item.isChecked = false;
+				await updateItem(listToken, item);
+			}
+		} catch (error) {
+			console.log('error', error);
+		}
+	};
+
+	useEffect(() => {
+		if (timeElapsed > one_day_in_ms) {
+			item.isChecked = false;
+			updateItem(listToken, item);
+		}
+	}, []);
```

Comment:
There is an eslint error when the application loads up.

```
  40:5  warning  React Hook useEffect has missing dependencies: 'item', 'listToken', 'one_day_in_ms', and 'timeElapsed'. Either include them or remove the dependency array  react-hooks/exhaustive-deps
```

Currently an empty array, `[]` is there. If there really are no dependencies, remove the empty array as the eslint rule suggests.

Reply:
I have removed the dependency array.